os/pm/pm_recover.c

```diff
@@ -0,0 +1,54 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/pm/pm.h>
+
+#include "pm.h"
+
+#ifdef CONFIG_PM
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: pm_recover
+ *
+ * Description:
+ *   This function is called from task_recover() when a task is deleted via
+ *   task_delete() or via pthread_cancel(). It do pm post cleaning.
+ *
+ * Inputs:
+ *   tcb - The TCB of the terminated task or thread
+ *
+ * Return Value:
+ *   None.
+ *
+ ****************************************************************************/
+void pm_recover(FAR struct tcb_s *tcb)
+{
+#ifdef CONFIG_PM_METRICS
+    pm_metrics_update_recover(tcb->pid);
```

Comment:
we can task exited using PIDHASH is same

For only pm_metrics, do we need to add operation in task exit?

Reply:
Yes, but in future we can use `pm_recover` for other pm cleanup activities which may be require after task termination.

Reply:
I think, when add new future, add this also is better.

for pm_metrics side, It seems better to keep the data until the next thread is created.

Reply:
If we do this way, then we may need to keep `thread_name` in memory. Can we do this?